src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1355,12 +1441,12 @@ internal static void ValidateFunction(FunctionDescriptor function, Dictionary<st
 
                 // prevent duplicate/conflicting routes
                 foreach (var pair in httpFunctions)
+            {
+                if (HttpRoutesConflict(httpTrigger, pair.Value))
                 {
-                    if (HttpRoutesConflict(httpTrigger, pair.Value))
-                    {
-                        throw new InvalidOperationException($"The route specified conflicts with the route defined by function '{pair.Key}'.");
-                    }
+                    throw new InvalidOperationException($"The route specified conflicts with the route defined by function '{pair.Key}'.");
```

Comment:
Currently functions ignores the route prefix in this check. which should be a bug. ([https://github.com/Azure/azure-webjobs-sdk-script/issues/1781](https://github.com/Azure/azure-webjobs-sdk-script/issues/1781))

Proxies by design do not honor route prefix. The implication of this is. Lets say you have a function with a route /api/ip (/api is the route prefix by default). An now you added a proxy "/ip". In the previous code, this would generate an error. But this should be allowed, and is also a way people use proxies. additionally, there is a duplicate check in the proxies dll. For cases where proxies and function share the same route (considering route prefix). For. eg: proxy is defined as "/api/ip" the function will be ignored. Once the bug is fixed in functions, we can enable checks for duplicate between functions and proxies.

---
In reply to: [132541071](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r132541071) [](ancestors = 132541071)